File: 02- Estructuras de control/java/ConsumoElectrico.java
/**
 * Descripción: Una compañía de energía eléctrica desea desarrollar un programa que permita calcular el monto a pagar de un usuario doméstico.
 * Autor: Santiago De la mora
 * Fecha: 27/01/2026
 */

import javax.swing.JOptionPane;

public class ConsumoElectrico {
    public static void main(String[] args) {
        // Declaración e inicialización de variables
        double consumoMensual = 0.0,
                costoPorKw = 0.0,
                recargoPorcentaje = 0.0,
                recargo = 0.0,
                monto = 0.0,
                montoFinal = 0.0;
        String tipoTarifa, mensaje;

        // Entrada
        consumoMensual = Double.parseDouble(
                JOptionPane.showInputDialog(
                        null,
                        "Ingrese su consumo mensual en KiloWatts-hora (KWh)",
                        "Consumo mensual",
                        JOptionPane.PLAIN_MESSAGE
                )
        );

        // Validación
        if (consumoMensual < 0) {
            JOptionPane.showMessageDialog(
                    null,
                    "No puede ingresar un consumo KiloWatt-hora negativo.",
                    "Error",
                    JOptionPane.ERROR_MESSAGE
            );
        } else {
            // Entrada
            tipoTarifa = JOptionPane.showInputDialog(
                    null,
                    "Ingrese el tipo de tarifa\n1) Básica\n2) Intermedia\n3) Alta",
                    "Error",
                    JOptionPane.PLAIN_MESSAGE
            );

            // Verifca el tipo de tarifa
            switch (tipoTarifa) {
                case "1":
                    tipoTarifa = "Básica";
                    costoPorKw = 0.85;

                    // Aplica recargoPorcentaje si se cumple la condición
                    if (consumoMensual > 250)
                        recargoPorcentaje = 0.12;

                    break;

                case "2":
                    tipoTarifa = "Intermedia";
                    costoPorKw = 1.25;

                    // Aplica recargoPorcentaje si se cumple la condición
                    if (consumoMensual >= 300 && consumoMensual <= 500)
                        recargoPorcentaje = 0.10;
                    else if (consumoMensual > 500)
                        recargoPorcentaje = 0.18;

                    break;

                case "3":
                    tipoTarifa = "Alta";
                    costoPorKw = 2.10;

                    // Aplica recargoPorcentaje si se cumple la condición
                    if (consumoMensual > 400)
                        recargoPorcentaje = 0.25;

                    break;

                default:
                    JOptionPane.showMessageDialog(
                            null,
                            "Sólo puede ingresar una opción de el 1 al 3.",
                            "Error",
                            JOptionPane.ERROR_MESSAGE
                    );

                    return;
            }

            // Calcula el precio a pagar
            monto = consumoMensual * costoPorKw;
            recargo = (monto * recargoPorcentaje);
            montoFinal = monto + recargo;

            // Salida
            mensaje = String.format("""
                Tipo de tarifa aplicada: %s
                Consumo registrado: %.2f kWh
                Costo por kWh: $%.2f
                Recargos aplicados: $%.2f
                Total a pagar: $%.2f\n\n""",
                tipoTarifa, consumoMensual, costoPorKw, recargo, montoFinal) +
                ((montoFinal > 1500.0) ? "Consumo elevado: se recomienda ahorro de energía." : "") +
                ((consumoMensual < 100) ? "Usuario con consumo eficiente." : "");

            JOptionPane.showMessageDialog(null, mensaje, "Recibo de luz", JOptionPane.PLAIN_MESSAGE);
        }
    }
}

File: 02- Estructuras de control/python/consumo_electrico.py
"""
Descripción: Una compañía de energía eléctrica desea desarrollar un programa que permita calcular el monto a pagar de un usuario doméstico.
Autor: Santiago De la mora
Fecha: 01/02/2026
"""

# Declaración e inicialización de variables
consumo_mensual = 0.0
costo_por_kw = 0.0
recargo_porcentaje = 0.0
recargo = 0.0
monto = 0.0
monto_final = 0.0
tipo_tarifa = ""
mensaje = ""

# Entrada
consumo_mensual = float(input("Ingrese su consumo mensual en KiloWatts-hora (KWh):\n"))

# Validación
if consumo_mensual < 0:
    print("\nNo puede ingresar un consumo KiloWatt-hora negativo.\n")
else:
    # Entrada
    tipo_tarifa = input("\nIngrese el tipo de tarifa\n1) Básica\n2) Intermedia\n3) Alta\n")

    # Verifica el tipo de tarifa
    match tipo_tarifa:
        case "1":
            tipo_tarifa = "Básica"
            costo_por_kw = 0.85

            # Aplica el recargo si se cumple la condición
            if consumo_mensual > 250:
                recargo_porcentaje = 0.12

        case "2":
            tipo_tarifa = "Intermedia"
            costo_por_kw = 1.25

            # Aplica el recargo si se cumple la condición
            if consumo_mensual >= 300 and consumo_mensual <= 500:
                recargo_porcentaje = 0.10
            elif consumo_mensual > 500:
                recargo_porcentaje = 0.18

        case "3":
            tipo_tarifa = "Alta"
            costo_por_kw = 2.10

            # Aplica el recargo si se cumple la condición
            if consumo_mensual > 400:
                recargo_porcentaje = 0.25

        case _:
            print("\nSólo puede ingresar una opción del 1 al 3.\n")

    # Calcula el precio a pagar
    monto = consumo_mensual * costo_por_kw
    recargo = (monto * recargo_porcentaje)        
    monto_final = monto + recargo

    # Salida
    mensaje = f"""
Tipo de tarifa aplicada: {tipo_tarifa}
Consumo registrado: {consumo_mensual:.2f} kWh
Costo por kWh: ${costo_por_kw:.2f}
Recargos aplicados: ${recargo:,.2f}
Total a pagar: ${monto_final:,.2f}"""
    
    # Agrega los mensajes especiales a la cadena si corresponde
    mensaje += "\n\nConsumo elevado se recomienda ahorro de energía." if (monto_final > 1500.0) else ""
    mensaje += "\n\nUsuario con consumo eficiente." if (consumo_mensual < 100) else ""

    print(mensaje)